src/core/compiler.jl

```diff
@@ -252,6 +304,19 @@ function build_model_info(input_expr)
     return model_info
 end
 
+function to_nt_expr(syms::Vector, vals = syms)
```

Comment:
Maybe change to a more telling name? `tonamedtuple_expr` ?

Reply:
Sure.